### ISSUE TITLE
Treat numbers in IRF table columns as `ndmin=1`

### ISSUE THREAD

**Issue by maxnoe** [gammapy/gammapy#3024]
It is currently impossible using astropy to read or write 1d arrays with a single entry.
They will be written and read as single numbers instead of 1d arrays of length 1.

This means, that IRFs cannot be written with single bins in any of the dimensions, which is most important for the FOV offset angle for point-like IRFs.

Issue at astropy: https://github.com/astropy/astropy/issues/10765

Gammapy should treat single numbers in columns like `THETA_LOW`, `THETA_HI` as 1d arrays with a single entry, e.g. by calling
something like `u.Quantity(table['THETA_LO'], ndmin=1)`.

CUrrently this fails due to len being called on a scalar:

```
In [5]: EffectiveAreaTable2D.read('./pyirf_eventdisplay.fits.gz', hdu='EFFECTIVE_AREA')
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-5-1f951cbdf154> in <module>
----> 1 EffectiveAreaTable2D.read('./pyirf_eventdisplay.fits.gz', hdu='EFFECTIVE_AREA')

~/.local/anaconda/envs/pyirf-dev/lib/python3.7/site-packages/gammapy/irf/effective_area.py in read(cls, filename, hdu)
    413         """Read from file."""
    414         with fits.open(make_path(filename), memmap=False) as hdulist:
--> 415             return cls.from_hdulist(hdulist, hdu=hdu)
    416 
    417     def to_effective_area_table(self, offset, energy=None):

~/.local/anaconda/envs/pyirf-dev/lib/python3.7/site-packages/gammapy/irf/effective_area.py in from_hdulist(cls, hdulist, hdu)
    407     def from_hdulist(cls, hdulist, hdu="EFFECTIVE AREA"):
    408         """Create from `~astropy.io.fits.HDUList`."""
--> 409         return cls.from_table(Table.read(hdulist[hdu]))
    410 
    411     @classmethod

~/.local/anaconda/envs/pyirf-dev/lib/python3.7/site-packages/gammapy/irf/effective_area.py in from_table(cls, table)
    401             offset_hi=table["THETA_HI"].quantity[0],
    402             data=table["EFFAREA"].quantity[0].transpose(),
--> 403             meta=table.meta,
    404         )
    405 

~/.local/anaconda/envs/pyirf-dev/lib/python3.7/site-packages/gammapy/irf/effective_area.py in __init__(self, energy_lo, energy_hi, offset_lo, offset_hi, data, meta, interp_kwargs)
    369             offset_axis = MapAxis.from_nodes(offset_lo, interp="lin", name="offset")
    370         else:
--> 371             offset_edges = edges_from_lo_hi(offset_lo, offset_hi)
    372             offset_axis = MapAxis.from_edges(offset_edges, interp="lin", name="offset")
    373 

~/.local/anaconda/envs/pyirf-dev/lib/python3.7/site-packages/gammapy/maps/utils.py in edges_from_lo_hi(edges_lo, edges_hi)
    153     edges = edges_lo.copy()
    154     try:
--> 155         edges = edges.insert(len(edges), edges_hi[-1])
    156     except AttributeError:
    157         edges = np.insert(edges, len(edges), edges_hi[-1])

~/.local/anaconda/envs/pyirf-dev/lib/python3.7/site-packages/astropy/units/quantity.py in __len__(self)
   1064         if self.isscalar:
   1065             raise TypeError("'{cls}' object with a scalar value has no "
-> 1066                             "len()".format(cls=self.__class__.__name__))
   1067         else:
   1068             return len(self.value)

TypeError: 'Quantity' object with a scalar value has no len()

```

**Comment by adonath** [comment]
Thanks @MaxNoe! I think this minimal example reproduces the issue:

```python3
from gammapy.irf import EffectiveAreaTable2D
from astropy.table import Table
from astropy import units as u

table = Table()

table["ENERG_LO"] = [1, 3] * u.TeV
table["ENERG_HI"] = [3, 10] * u.TeV
table["THETA_LO"] = 1 * u.deg
table["THETA_HI"] = 1 * u.deg
table["EFFAREA"] = [[1, 1]]

aeff = EffectiveAreaTable2D.from_table(table)
```

I agree we should implement a fix for this on the Gammapy side, as the fix in Astropy might take a while. I'll think a bit about where the best place for the fix is and make a proposal here later.

**Comment by MaxNoe** [comment]
ACtually the fix is about to be merged and should be released with astropy 4.0.2

**Comment by chaimain** [comment]
Hi @adonath, this issue is resolved and released in astropy 4.0.2. Will the fix in gammapy be a straight-forward change or a bit time-consuming modification? I can try if it is a simple fix.

**Comment by maxnoe** [comment]
Gammapy does not need to fix anything if you update to astropy 4.0.2.

Gammapy could fix it for astropy < 4.0.2. But I wouldn't, since these files are actually not standard compliant.

**Comment by maxnoe** [comment]
@adonath since the fix is now published in astropy 4.0.2, I will close this as not a gammapy issue.

If you want to support older gammapy versions, you could implement something, but I wouldn't.
